File: projectEuler/p3.py
#!/usr/bin/env python3
"""
problem 3; Largest prime number
"""

from math import sqrt

def is_prime(num):
    for i in range(2, int(sqrt(num))):
        if num % i == 0:
            return False
    return True

def prime_factors(num):
    prime_factors = []
    
    for i in range(2, int(sqrt(num))):
        if num % i == 0:
            if is_prime(i):
                prime_factors.append(i)
            if is_prime(num / i):
                prime_factors.append(i)
    print("Largest prime factor is {}".format(max(prime_factors)))

prime_factors(600851475143)

File: projectEuler/p3.c
#include<stdio.h>

/**
 * main - Problem 3
 *
 * Description: Largest prime factor
 *
 * Return: Always 0 (Success)
*/

int main(void)
{
	long long num = 600851475143;
	int div = 2, maxPrime = 0;

	// loop through all prime factors both even and odd
	for (div = 2; num != 1; div += 2)
	{
		while (num % div == 0)
		{
			maxPrime = div;
			num /= div;
		}

		// when done looping on even prime number 2 increament by 1
		if (div == 2 && num % div != 0)
			++div;
	}

	if (maxPrime > 0)
		printf("Largest prime factor is %d\n", maxPrime);

	return (0);
}
